cocos/ui/UIRichText.cpp

```diff
@@ -1786,6 +1818,13 @@ void RichText::formatRenderers()
     }
     else
     {
+        // erase empty rears
+        while (_elementRenders.back().empty())
+        {
+            _elementRenders.pop_back();
+        }
```

Comment:
When element render will be empty?

Reply:
This is the case of end with some new lines, for example, the **last** element of a `RichText` is `RichElementNewLine`, or `RichElementText` with some `\n`'s.
`RichText::addNewLine()` will be invoked, but no renders in the vector.

Reply:
Why remove them? May be it is the expected effect, use some new line the make some space at the end of content.

Reply:
At first, I think new lines at end of RichText is meaningless.
Maybe we can add a method to decide whether trim or not, how about it?

Reply:
I think it may make things complex. If they don't want the effect, then why add these new lines explicitly?

Reply:
OK. I see. I will delete these codes